sandy/fy.py

```diff
@@ -422,23 +422,32 @@ def apply_qmatrix(self, zam, energy, decay_data):
         >>> decay_minimal = sandy.get_endf6_file("jeff_33", 'decay', zam)
         >>> decay_fytest = sandy.DecayData.from_endf6(decay_minimal)
         >>> npfy = Fy(minimal_fytest_2)
-        >>> npfy_pert = npfy.apply_bmatrix(942390, 5.00000e+05, decay_fytest)
-        >>> diff = npfy_pert.data[npfy_pert.data.FY != npfy.data.FY].FY
-        >>> comp = npfy_pert.data.query('ZAM==942390 & MT==459 & E==500e3').squeeze().FY
-        >>> assert comp.values.all() == diff.values.all()
-
-        """
-        new_data = self.data.copy()
+        >>> npfy_pert = npfy.apply_qmatrix(942390, 5.00000e+05, decay_fytest)
+        >>> npfy_pert.data[npfy_pert.data.MT == 459]
+             MAT	 MT	   ZAM	   ZAP	          E	         FY	        DFY
+        3	9437	459	942390	591480	5.00000e+05	1.00000e-01	0.00000e+00
+        4	9437	459	942390	591481	5.00000e+05	2.00000e-01	0.00000e+00
+        5	9437	459	942390	601480	5.00000e+05	6.00000e-01	0.00000e+00
+        6	9437	459	942390	621480	5.00000e+05	6.00000e-01	0.00000e+00
+        """
+        # Obtain the data:
+        data = self.data.copy()
         conditions = {'ZAM': zam, 'MT': 454, "E": energy}
-        fy_data = self._filters(conditions).data.set_index('ZAP')['FY']
-        index = fy_data.index
+        fy_data = self._filters(conditions).data
+        mat = fy_data.iloc[0, 0]
+        fy_data = fy_data.set_index('ZAP')['FY']
         Q = decay_data.get_qmatrix()
+        # Put the data in a approppiate format:
+        mask = (data.ZAM == zam) & (data.MT == 459) & (data.E == energy)
+        data = pd.concat([data, data.loc[mask]]).drop_duplicates(keep=False)
```

Comment:
why are you doing this?

Reply:
To remove the duplicate data

Reply:
what duplicate data? why do you need concat?

Reply:
To remove only data.loc[mask].

Reply:
I still don't get it. `data.loc[mask]` is already in `data`, then why do you duplicate it using `concat`? and why then you drop the duplicates?

Reply:
Because is the only way that i found in internet to drop from a dataFrame another dataFrame

Reply:
but why do you want to do it? what do you want to do? if you completely delete this line the code would still work fine, right? or not?

Reply:
No, in this line i am droping all cumulative fission yields defined by the mask. Then, when i use concat, the enter because the data that is suppose to be replacing is not there

Reply:
what is `pd.concat([data, data.loc[mask]])` doing? it concatenates duplicates of the CFY selected by the mask, right? those CFYs already existed in data

Reply:
The problem was that i was not able to erase the row containing the information of data.loc[mask]]. So i found in the Internet the following method. First fetch the two data, duplicating CFY selected by the mask. Then, with drop_duplicates the duplicated files will be erased. 
In this way, after this line, the data doesnt have the information of the CFY selected by the mask

Reply:
Now I see what you are trying to do...
It should be something like `data = data.loc[~mask]`

Reply:
Yes !!!! I changed now, thanks